src/Microsoft.Extensions.Caching.Abstractions/CacheEntryExtensions.cs

```diff
@@ -148,7 +136,6 @@ public static ICacheEntry SetOptions(this ICacheEntry entry, MemoryCacheEntryOpt
             entry.AbsoluteExpiration = options.AbsoluteExpiration;
             entry.AbsoluteExpirationRelativeToNow = options.AbsoluteExpirationRelativeToNow;
             entry.SlidingExpiration = options.SlidingExpiration;
-            entry.Priority = options.Priority;
```

Comment:
Removing Priority is unfortunate but I understand. Do we have another way to add custom metadata for the eviction algorithm without embedding it in the value itself? What if ICacheEntry had a transparent `object EvictionMetadata` property?